pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/distinct/BaseOffHeapDistinctSet.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.aggregation.function.distinct;
+
+import it.unimi.dsi.fastutil.HashCommon;
+import java.util.Iterator;
+import javax.annotation.concurrent.NotThreadSafe;
+import org.apache.pinot.segment.spi.memory.unsafe.Unsafer;
+import sun.misc.Unsafe;
+
+
```

Comment:
These classes would really use some javadocs. 
They seem to be fastutil sets adapted to store data off heap but what they store is actually HashCommon.mix() of the given values, most likely to avoid re-computing hashes when set grows.